packages/fixie/src/agent.ts

```diff
@@ -260,6 +262,39 @@ export class FixieAgent {
     return config as AgentConfig;
   }
 
+  private static inferRuntimeParametersSchema(agentPath: string): TJS.Definition | null {
+    // If there's a tsconfig.json file, try to use Typescript to produce a JSON schema
+    // with the runtime parameters for the agent.
+    const tsconfigPath = path.resolve(path.join(agentPath, 'tsconfig.json'));
+    if (!fs.existsSync(tsconfigPath)) {
+      term.yellow('⚠️ tsconfig.json not found. Your agent will not support runtime parameters.\n');
+      return null;
+    }
+
+    const settings: TJS.PartialArgs = {
+      required: true,
+      noExtraProps: true,
+    };
+
+    // We're currently assuming the entrypoint is exported from src/index.{ts,tsx}.
+    const handlerPath = path.resolve(path.join(agentPath, 'src/index.js'));
+    const tempPath = path.join(fs.mkdtempSync(path.join(os.tmpdir(), 'fixie-')), 'extract-parameters-schema.mts');
+    fs.writeFileSync(
+      tempPath,
+      `
+      import Handler from '${handlerPath}';
+      export type RuntimeParameters = Parameters<typeof Handler>[0] extends infer T ? T : {};
+      `
+    );
+    const program = TJS.programFromConfig(tsconfigPath, [tempPath]);
+    const schema = TJS.generateSchema(program, 'RuntimeParameters', settings);
+    if (schema && schema.type !== 'object') {
```

Comment:
```suggestion
    if (schema?.type !== 'object') {
```

Reply:
Actually, this is intentional -- `null !== "object"`